editor/utils/editor-can-upload/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Response cache of response header.
+ *
+ * @type {Object}
+ */
+export const cachedMediaHeader = {};
+
+/**
+ * Whether a request is running.
+ *
+ * @type {Boolean}
+ */
+let isRequestingMediaOptions = false;
+
+export function requestMediaAllowHeader( request ) {
+	isRequestingMediaOptions = true;
+
+	const promise = wp.apiRequest( request )
+		.then( ( body, status, xhr ) => {
+			return xhr.getResponseHeader( 'allow' );
+		} )
+		.then( ( response ) => {
```

Comment:
Does this need to be a separate `Promise#then`, or should it just be included in the one prior? If they're meant to be distinct steps, could we organize them as such with named functions?